<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Evaluation strategies &mdash; NetworkCommons alpha documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="_static/css/custom.css?v=2155bac8" />

  
    <link rel="shortcut icon" href="_static/nc_logo.svg"/>
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="_static/documentation_options.js?v=1f169f65"></script>
        <script src="_static/doctools.js?v=9a2dae69"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="_static/copybutton.js?v=f281be69"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="General guidelines for contribution" href="guidelines/landing.html" />
    <link rel="prev" title="Methods" href="methods.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="contents.html">
            
              <img src="_static/nc_logo.svg" class="logo" alt="Logo"/>
          </a>
              <div class="version">
                0.5.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Main</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="index.html">Bridging the gap between data, methods and knowledge in the network inference field</a></li>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="api.html">API</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Details</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="datasets.html">Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="methods.html">Methods</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Evaluation strategies</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#offtarget-recovery">Offtarget recovery</a></li>
<li class="toctree-l2"><a class="reference internal" href="#phosphorylation-sensitivity-to-drug-perturbation">Phosphorylation sensitivity to drug perturbation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#pathway-enrichment-analysis">Pathway enrichment analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#recovery-of-dysregulated-kinases">Recovery of dysregulated kinases</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Contribution guidelines</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="guidelines/landing.html">General guidelines for contribution</a></li>
<li class="toctree-l1"><a class="reference internal" href="guidelines/guide_1_data.html">Contribution’s guideline: Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="guidelines/guide_2_methods.html">Contribution’s guideline: Methods</a></li>
<li class="toctree-l1"><a class="reference internal" href="guidelines/guide_3_eval.html">Contribution’s guideline: Evaluation strategies</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Vignettes</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="vignettes/1_quickstart.html">Vignette 1: Introduction to NetworkCommons</a></li>
<li class="toctree-l1"><a class="reference internal" href="vignettes/2_multiple_methods.html">Vignette 2: Using multiple methods to infer the networks</a></li>
<li class="toctree-l1"><a class="reference internal" href="vignettes/3_evaluation_offt_path.html">Vignette 3: Evaluating methods under two scenarios: offtarget recovery and pathway activity</a></li>
<li class="toctree-l1"><a class="reference internal" href="vignettes/4_evaluation_decryptm.html">Vignette 4: Sensitive response to drug perturbation using phosphoproteomics</a></li>
<li class="toctree-l1"><a class="reference internal" href="vignettes/5_cptac_phosphoactivity.html">Vignette 5: Multi-omics validation - Recovery of dysregulated kinases in response to cancer mutations</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Additional resources</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="vignettes/A_moon.html">Vignette A: recursive propagation with MOON</a></li>
<li class="toctree-l1"><a class="reference internal" href="vignettes/B_pertpy.html">Vignette B: Pertpy bridging to NetworkCommons</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="contents.html">NetworkCommons</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="contents.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Evaluation strategies</li>
      <li class="wy-breadcrumbs-aside">
              <a href="https://github.com/saezlab/networkcommons/blob/master/docs/src/benchmarks.rst" class="fa fa-github"> Edit on GitHub</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="evaluation-strategies">
<h1>Evaluation strategies<a class="headerlink" href="#evaluation-strategies" title="Link to this heading"></a></h1>
<p>One of the main aims of NetworkCommons is to provide a comprehesive set of metrics to evaluate the performance of the different network inference methodologies.
Currently, we provide three different evaluation strategies.
If you want to contribute with your own, please check our <a class="reference internal" href="guidelines/guide_3_eval.html"><span class="doc">Contribution guidelines</span></a>.</p>
<section id="offtarget-recovery">
<span id="eval-offtarget"></span><h2>Offtarget recovery<a class="headerlink" href="#offtarget-recovery" title="Link to this heading"></a></h2>
<p><strong>Data</strong>: perturbational scenarios, e.g a drug perturbation, for which there are differential expression profiles between control and drug-perturbed samples.
See <a class="reference internal" href="datasets.html#details-panacea"><span class="std std-ref">PANACEA</span></a>.</p>
<p><strong>Assumption</strong>: In this setting, we assume that, in a perturbational context, the effects that are measured via omics data is not only a product of the perturbation origin
(e.g KO, KD, drug perturbation), but also of other origins of perturbation that are not directly targeted by the perturbation agent (e.g a drug offtarget).</p>
<object type="image/svg+xml" data="_static/nc_offtarget.svg" width="1000px" alt="Evaluation based on offtarget recovery"></object><p><strong>Performance metric:</strong> Share (%) of offtargets recovered in the solution network</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Methods that recover a higher share of offtargets, compared to a random control, will be more successful in contextualising the perturbation, since the method incorporates
the offtargets’ effect.</p>
</div>
<p><strong>Example:</strong> <span class="xref std std-doc">Vignette 1: A simple example</span></p>
</section>
<section id="phosphorylation-sensitivity-to-drug-perturbation">
<span id="eval-ec50"></span><h2>Phosphorylation sensitivity to drug perturbation<a class="headerlink" href="#phosphorylation-sensitivity-to-drug-perturbation" title="Link to this heading"></a></h2>
<p><strong>Data</strong>: phosphoproteomics dose-response curves, EC50 values, time-course data,
See <a class="reference internal" href="datasets.html#details-decryptm"><span class="std std-ref">DecryptM</span></a>.</p>
<p><strong>Assumption</strong>: In this setting, we assume that, in a perturbational context, those elements in a network that respond quicker to a perturbation (have a lower EC50) will be more
important in the contextualisation of said perturbation</p>
<p><strong>Performance metric:</strong> EC50 values for nodes included and excluded of the solution network.</p>
<object type="image/svg+xml" data="_static/nc_ec50.svg" width="1000px" alt="Evaluation based on sensitivity to drug perturbation"></object><div class="admonition note">
<p class="admonition-title">Note</p>
<p>Methods producing result networks whose nodes have a low average EC50 (compared to nodes not included in the network) are better performers than those producing networks
where this difference (EC50_in - EC50_out) is not that big.</p>
</div>
<p><strong>Example:</strong> <span class="xref std std-doc">Vignette 3: Sensitive response to drug perturbation using phosphoproteomics</span></p>
</section>
<section id="pathway-enrichment-analysis">
<span id="eval-pathway"></span><h2>Pathway enrichment analysis<a class="headerlink" href="#pathway-enrichment-analysis" title="Link to this heading"></a></h2>
<p><strong>Data</strong>: perturbational scenarios, dysregulation (e.g cancer basal profiles), basal profiles (e.g tissue specific profiles)
See <a class="reference internal" href="datasets.html#details-decryptm"><span class="std std-ref">PANACEA</span></a>.</p>
<p><strong>Assumption</strong>: In this setting, we use the nodes of the subnetworks to perform Overrepresentation Analysis against a set of predefined gene sets, among which we expect one to be especially represented
(for example, a specific pathway will be overrepresented if said pathway is perturbed, or is especially active/inactive in a given profile)</p>
<p><strong>Performance metric:</strong> rank of the selected gene set among all gene sets, according to ORA score</p>
<object type="image/svg+xml" data="_static/nc_pathway.svg" width="1000px" alt="Evaluation based on pathway enrichment"></object><div class="admonition note">
<p class="admonition-title">Note</p>
<p>Having preselected a gene set of interest, the methods producing networks in which the said geneset is ranked high, according to their ORA score, against others, will have a better performance.</p>
</div>
<p><strong>Example:</strong> <span class="xref std std-doc">Vignette 1: A simple example</span></p>
</section>
<section id="recovery-of-dysregulated-kinases">
<span id="eval-multiomics"></span><h2>Recovery of dysregulated kinases<a class="headerlink" href="#recovery-of-dysregulated-kinases" title="Link to this heading"></a></h2>
<p><strong>Data</strong>: perturbational scenarios, dysregulation (e.g cancer basal profiles)
See <a class="reference internal" href="datasets.html#details-cptac"><span class="std std-ref">CPTAC</span></a>.</p>
<p><strong>Assumption</strong>: In this setting, we use three different types of omics data:</p>
<ul class="simple">
<li><p><strong>Proteomics</strong>: we identified the most differentially abundant receptors in the proteomics profiles between healthy and tumor samples. We assume that if they are differentially abundant, they will be activated/inhibited.</p></li>
<li><p><strong>Transcriptomics</strong>: we performed TF enrichment analysis, in order to get the TFs that are dysregulated in the tumor samples compared to the healthy control.</p></li>
<li><p><strong>Phosphoproteomics</strong>: we performed kinase activity estimation and then evaluate the level of dysregulation in the resulting subnetwork.</p></li>
</ul>
<p><strong>Performance metric:</strong> difference between kinase activity score in the solution network and the overall PKN.</p>
<object type="image/svg+xml" data="_static/nc_multiomics.svg" width="1000px" alt="Evaluation based on sensitivity to drug perturbation"></object><div class="admonition note">
<p class="admonition-title">Note</p>
<p>Methods whose result subnetworks have an average higher kinase activity score, compared to the overall PKN, will be better performers.</p>
</div>
<p><strong>Example:</strong> <span class="xref std std-doc">Vignette 4: Recovery of dysregulated kinases in response to cancer mutations</span></p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="methods.html" class="btn btn-neutral float-left" title="Methods" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="guidelines/landing.html" class="btn btn-neutral float-right" title="General guidelines for contribution" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, NetworkCommons developers.</p>
  </div>

   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>